web/client/src/util/apiWrapper.js

```diff
@@ -19,3 +19,28 @@ export const validateToken = (requestOptions) => {
     return JSON.parse(JSON.stringify({ Error: error }));
   });
 };
+
+export const getHTTPResponse = (values) => {
+  const param = {
+    url: values.URI,
+    method: values.httpMethod.toLowerCase(),
+    headers: {
+      Authorization: "Bearer " + values.token,
+      ...(values.contentType ? { "Content-Type": values.contentType } : {}),
+      ...(values.reqBody ? { data: values.reqBody } : {}),
+    },
+  };
+  let i;
+  for (i = 0; i < values.headers.length; i++) {
+    if (
+      values.headers[i].headerName.length !== 0 &&
+      values[i].headerValue.length !== 0
+    ) {
+      param.headers[values.headers[i].headerName] =
+        values.headers[i].headerValue;
+    }
+  }
```

Comment:
This validation should probably be done in the request module and prevent a user from making the request until the conditions are met.

Reply:
moved to request module